aten/src/ATen/FunctionalTensorWrapper.cpp

```diff
@@ -788,6 +788,11 @@ void mutate_view_meta(const at::Tensor& self, const functionalization::ViewMeta&
 // The output meta tensor's stride info serves as a reference for what the correct strides should be.
 void set_sizes_strides_offset(const Tensor& out, const Tensor& reference_out) {
   out.unsafeGetTensorImpl()->set_sizes_and_strides(reference_out.sym_sizes(), reference_out.sym_strides(), reference_out.sym_storage_offset());
+
+  TORCH_INTERNAL_ASSERT(at::functionalization::impl::isFunctionalTensor(out));
+  auto value = from_functional_tensor(out);
+
+  value.unsafeGetTensorImpl()->set_sizes_and_strides(reference_out.sym_sizes(), reference_out.sym_strides(), reference_out.sym_storage_offset());
```

Comment:
I think I would prefer to only run this code for XLA/Lazy backends, just so it doesn't mess with the default compile path

Reply:
(also, as you pointed out - if we can get this to pass CI you can probably also kill the set_sizes_strides on line 790, since it should be a no-op)